packages/terra-functional-testing/src/reporters/fileOutput-reporter/index.js

```diff
@@ -0,0 +1,220 @@
+const fs = require('fs-extra');
+const path = require('path');
+const stripAnsi = require('strip-ansi');
+const SpecReporter = require('@wdio/spec-reporter').default;
+const endOfLine = require('os').EOL;
+const { Logger } = require('@cerner/terra-cli');
+
+const LOG_CONTEXT = '[Terra-Toolkit:fileOutput-reporter]';
+
+class FileOutputReporter extends SpecReporter {
+  constructor() {
+    super();
+    this.runners = [];
+    this.resultJsonObject = {
+      startDate: '',
+      type: 'wdio',
+      locale: '',
+      formFactor: '',
+      theme: '',
+      output: {},
+      endDate: '',
+    };
+    this.fileName = '';
+    this.moduleName = process.cwd().split(path.sep).pop();
+
+    this.setResultsDir = this.setResultsDir.bind(this);
+    this.hasResultsDir = this.hasResultsDir.bind(this);
+    this.setTestModule = this.setTestModule.bind(this);
+    this.printReport = this.printReport.bind(this);
+    this.getMessage = this.getMessage.bind(this);
+
+    this.setResultsDir();
+    this.hasResultsDir();
+  }
+
+  /**
+   * Sets results directory for the test run. Uses the wdio reporterOptions.outputDir if set, otherwise
+   * it outputs to tests?/wdio/reports.
+   * @return null;
+   */
+  setResultsDir() {
+    const { reporterOptions } = this.options;
```

Comment:
can you walk through how `this.options` is being propagated and set? Is it coming from the spec reporter that we're extending? 

I tried to dig into this, and my understanding is when we specify what reporters to use inside configuration, we can pass an option as the second argument: `reporters: [[fileOutputReporter, { dir: 'full-stack' }]],`, and that options object will propagate here - is this totally wrong?

Reply:
yes you are right options come from the configurations, in this case we are not using them so, I removed that section

Reply:
https://github.com/cerner/terra-toolkit/pull/662/files#diff-5aca2141ee738df380d2de0da498a0a17727d935d42de74bbbb00a3046b1febeR41-R47